Extraction of chain configuration

tzdybal opened the pull request:
This will resolve #89.

tzdybal committed:
core: extract chain genesis config to files

All chains defined in multi-geth, and not defined in
ethereum/go-ethereum was extracted to separate files.

tzdybal committed:
core: add missing chains to `configOrDefault`

tzdybal committed:
params: extract chain configs to separate files

tzdybal committed:
params: bootnodes extracted to per-chain config

tzdybal committed:
consensus,mobile,params: more per-chain extraction

tzdybal committed:
consensus,params: extraction of more variables

tzdybal committed:
mobile: missing imports

tzdybal committed:
consensus: chain specific logic extracted

meowsbits commented on params/config_classic.go:
```diff
@@ -0,0 +1,73 @@
+
+	// ClassicBootnodes are the enode URLs of the P2P bootstrap nodes running on
+	// the Ethereum Classic network.
+	ClassicBootnodes = []string{
```
Perhaps `*Bootnodes` vars should be in `params/bootnodes_*.go` files? (Just to echo the existing pattern).

tzdybal replied:
I was thinking about this, but then I thought that having separate file for a just 1 or 2 `var`s seems to be an overkill. Same for `genesis_alloc`.

meowsbits replied:
Roger, understood. I guess despite the overkill I'd prefer to see as close to a 'reflective iteration' of upstream patterns whenever possible just to emphasize structural consistency (as opposed to introducing sane arbitrations ;)  Also then I don't have to remember anything new when I try to look up Kotti bootnodes )

Not fervent or evangelical in my view though, just sayin'. LGTM as-is, too.

meowsbits replied:
Ha, on the flipside, though, files like this https://github.com/multi-geth/multi-geth/pull/88/files#diff-585fd2d517a40a26f4efdec5e4edf4ca will really push the fundamentalist overkiller envelope.

tzdybal replied:
10 consts - seems legit :rofl:

tzdybal committed:
params: chain bootnodes extracted to own files

tzdybal committed:
core: alloc extracted to own files

meowsbits committed:
core: rm core/genesis_test.go

This was accidentally not removed; file has
been effectively renamed to genesis_alloc_test.go
to avoid _chain namespace confusion.

meowsbits committed:
params: rm params/bootnodes_test.go

File contains redeclared fn
TestCheckCompatible from params/config_test.go
